test/puppetlabs/trapperkeeper/services/metrics/metrics_service_test.clj

```diff
@@ -58,7 +69,24 @@
          (is (= 200 (:status resp)))
          (doseq [[metric path] body
                  :let [resp (http-client/get (str "http://localhost:8180/metrics/v1" path))]]
-           (is (= 200 (:status resp))))))
+           (is (= 200 (:status resp)))))
+
+       (let [resp (http-client/get "http://localhost:8180/metrics/v2/list")
+             body (parse-response resp)]
+         (is (= 200 (:status  resp)))
+         (doseq [[namesp mbeans] (get body "value") mbean (keys mbeans)
+                 :let [url (str "http://localhost:8180/metrics/v2/read/"
+                                (jolokia-encode (str namesp ":" mbean))
+                                ;; NOTE: Some memory pools intentionally don't
+                                ;; implement MBean attributes. This results
+                                ;; in an error being thrown when those
+                                ;; attributes are read and is expected.
+                                "?ignoreErrors=true")
```

Comment:
Thanks for the explanation.  Makes sense that you did it this way to mirror what the v1 tests were doing.  Given that, I'm fine with this.